src/pages/nft/claim/index.vue

```diff
@@ -739,15 +742,38 @@ export default {
   async mounted() {
     const { redirect, free, from, state, ...query } = this.$route.query;
     let price;
-    try {
-      const { data } = await this.$api.get(
-        getNFTBookPaymentStatusEndpoint({
-          classId: this.classId,
-          collectionId: this.collectionId,
-          paymentId: this.paymentId,
-        })
-      );
-      ({ price } = data);
+    if (this.paymentId) {
+      try {
+        const { data } = await this.$api.get(
+          getNFTBookPaymentStatusEndpoint({
+            classId: this.classId,
+            collectionId: this.collectionId,
+            paymentId: this.paymentId,
+          })
+        );
+        ({ price } = data);
+        const {
+          giftInfo,
+          isPhysicalOnly,
+          status,
+          autoMemo,
+          isAutoDeliver,
+        } = data;
+        this.giftInfo = giftInfo;
+        this.isPhysicalOnly = isPhysicalOnly;
+        this.isAutoDeliver = isAutoDeliver;
+        this.creatorMessage = autoMemo;
+        this.status = status;
+        if (this.collectionId) {
+          await this.lazyFetchNFTCollectionInfo();
+        }
+      } catch (err) {
+        // eslint-disable-next-line no-console
+        console.error(err);
+      }
+    } else {
+      const { prices } = this.getNFTBookStoreInfoByClassId(this.classId);
+      const data = prices[this.priceIndex];
       const {
         giftInfo,
```

Comment:
Why we expect `giftInfo` and `status` exists in `NFTBookStoreInfoByClassId`? What are their expected values?

Reply:
I forgot that free books do not have the gifting feature, thanks 😝